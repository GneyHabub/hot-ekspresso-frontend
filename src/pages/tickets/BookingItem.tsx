import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  Box,
  Button,
  Card,
  colors,
  Divider,
  Stack,
  Typography,
} from '@mui/material';
import React from 'react';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import MoreHorizIcon from '@mui/icons-material/MoreHoriz';
import dayjs from 'dayjs';
import {
  Timeline, TimelineConnector, TimelineContent, TimelineDot, TimelineItem, TimelineSeparator,
} from '@mui/lab';
import { Booking } from '../../utils/types';

interface Props {
  booking: Booking;
  idx: number;
  onPick?: (b: Booking) => void;
}

const BookingItem: React.FC<Props> = ({ booking, idx, onPick }) => (
  <Accordion>
    <AccordionSummary
      expandIcon={<ExpandMoreIcon />}
      aria-controls={`panel${idx + 1}a-content`}
      id={`panel${idx + 1}a-content`}
    >
      <Box
        sx={{
          display: 'flex',
          alignItems: 'center',
          gap: '10px',
        }}
      >
        <Typography>{booking.flights[0].from}</Typography>
        <MoreHorizIcon />
        <Typography>{booking.flights[booking.flights.length - 1].to}</Typography>
      </Box>
      <Typography variant="overline" sx={{ ml: '10px' }}>
        {dayjs(booking.flights[0].departure).format('DD.MM.YYYY')}
      </Typography>
    </AccordionSummary>
    <AccordionDetails
      sx={{
        display: 'flex',
        gap: '5px',
      }}
    >
      <Box
        sx={{
          width: '20%',
          padding: '15px',
        }}
      >
        <Typography variant="subtitle1" fontSize="18px">Route</Typography>
        <Timeline>
          {booking.flights.map((flight) => (
            <TimelineItem key={flight.from}>
              <TimelineSeparator>
                <TimelineDot />
                <TimelineConnector />
              </TimelineSeparator>
              <TimelineContent>
                {flight.from}
              </TimelineContent>
            </TimelineItem>
          ))}
          <TimelineItem>
            <TimelineSeparator>
              <TimelineDot />
            </TimelineSeparator>
            <TimelineContent
              sx={{
                paddingBottom: 0,
              }}
            >
              {booking.flights[booking.flights.length - 1].to}
            </TimelineContent>
          </TimelineItem>
        </Timeline>
      </Box>
      <Divider orientation="vertical" flexItem />
      <Box
        sx={{
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'space-between',
          width: '100%',
        }}
      >
        <Box
          sx={{
            padding: '15px',
          }}
        >
          <Typography variant="subtitle1" fontSize="18px">
            <b>Price:</b>
            {' '}
            â‚¸
            {booking.price}
          </Typography>
          <Typography variant="subtitle1" fontSize="18px">
            <b>Departure:</b>
            {' '}
            {dayjs(booking.flights[0].departure).format('DD.MM.YYYY HH:mm')}
          </Typography>
          <Typography variant="subtitle1" fontSize="18px">
            <b>Arrival:</b>
            {' '}
            {dayjs(booking.flights[booking.flights.length - 1].arrival).format('DD.MM.YYYY HH:mm')}
          </Typography>
          <Typography variant="subtitle1" fontSize="18px">
            <b>Service Class:</b>
            {' '}
            {booking.serviceClass}
          </Typography>
        </Box>
        {onPick
              && (
              <Button
                variant="contained"
                onClick={() => onPick(booking)}
                sx={{
                  alignSelf: 'center',
                }}
              >
                Choose
              </Button>
              )}
      </Box>
    </AccordionDetails>
  </Accordion>
);

export default BookingItem;
