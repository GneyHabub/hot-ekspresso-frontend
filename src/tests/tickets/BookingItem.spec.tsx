import { LocalizationProvider } from '@mui/lab';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import BookingItem from '../../pages/tickets/BookingItem';

describe('BookingItem', () => {
  it('matches the snapshot', () => {
    const wrapper = render(
      <BrowserRouter>
        <BookingItem
          booking={{
            id: 'AAA',
            price: 300,
            flights: [{
              from: 'KZN',
              to: 'GSV',
              departure: new Date('1995-12-17T03:24:00'),
              arrival: new Date('1995-12-17T03:24:00'),
              carrier: 'RedWings',
            }],
            serviceClass: 'Economy',
          }}
          idx={1}
          onPick={() => {}}
        />
      </BrowserRouter>,
    );
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
