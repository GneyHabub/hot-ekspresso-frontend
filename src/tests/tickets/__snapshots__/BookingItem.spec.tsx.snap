// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingItem matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-controls="panel2a-content"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
        id="panel2a-content"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <div
            class="MuiBox-root css-1h2sv0j"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              KZN
            </p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MoreHorizIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              GSV
            </p>
          </div>
          <span
            class="MuiTypography-root MuiTypography-overline css-hsgdsv-MuiTypography-root"
          >
            17.12.1995
          </span>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel2a-content"
              class="MuiAccordion-region"
              id="panel2a-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-1owrxue-MuiAccordionDetails-root"
              >
                <div
                  class="MuiBox-root css-1vgca5b"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                  >
                    Route
                  </h6>
                  <ul
                    class="MuiTimeline-root MuiTimeline-positionRight css-55vwbm-MuiTimeline-root"
                  >
                    <li
                      class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-ha3bif-MuiTimelineItem-root"
                    >
                      <div
                        class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
                      >
                        <span
                          class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey css-v6mo1y-MuiTimelineDot-root"
                        />
                        <span
                          class="MuiTimelineConnector-root css-1ami7lv-MuiTimelineConnector-root"
                        />
                      </div>
                      <div
                        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-18y2uza-MuiTypography-root-MuiTimelineContent-root"
                      >
                        KZN
                      </div>
                    </li>
                    <li
                      class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-ha3bif-MuiTimelineItem-root"
                    >
                      <div
                        class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
                      >
                        <span
                          class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledGrey css-v6mo1y-MuiTimelineDot-root"
                        />
                      </div>
                      <div
                        class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1da6d3z-MuiTypography-root-MuiTimelineContent-root"
                      >
                        GSV
                      </div>
                    </li>
                  </ul>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1tk1bks-MuiDivider-root"
                />
                <div
                  class="MuiBox-root css-t6ot1t"
                >
                  <div
                    class="MuiBox-root css-1v1qs4"
                  >
                    <h6
                      class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                    >
                      <b>
                        Price:
                      </b>
                       â‚¸
                      300
                    </h6>
                    <h6
                      class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                    >
                      <b>
                        Departure:
                      </b>
                       
                      17.12.1995 03:24
                    </h6>
                    <h6
                      class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                    >
                      <b>
                        Arrival:
                      </b>
                       
                      17.12.1995 03:24
                    </h6>
                    <h6
                      class="MuiTypography-root MuiTypography-subtitle1 css-1s63ngw-MuiTypography-root"
                    >
                      <b>
                        Service Class:
                      </b>
                       
                      Economy
                    </h6>
                  </div>
                  <button
                    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1leidq1-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Choose
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
